### DISCUSSION TITLE
A scope mechanic for shebang recipes

### DISCUSSION THREAD

**Discussion by MostHated** [casey/just#1553] [Ideas]
Hey there, huge fan of 'just'. I just wanted to throw out a feature request for a rainy day. I was working with a shebang recipe yesterday and had some whitespace issues and thought it would be quite helpful to have a start/end scope marker, to allow for new lines.

An example could look something like this:

```justfile
# Note: The opening ({ and closing }) could be anything
# Whatever works best for parsing or whathave you.

# My Recipe Scope
my-recipe:
    #{{shebang}}
    ({
        Anything in here

        Gets sent to the temp script
    
        To be ran as a script
    });
```

This is obviously not a major need, just helps visually, to me at least. To avoid things like this if I wanted some separation:

```justfile
recipe:
    #{{shebang}}
    $var1 = "yo"
    $var2 = "also yo"
    #
    if($var1 == $var2) { echo "how in the even?" }
````

**Comment by runeimp** [comment]
In what situation are you running into whitespace issues? I have extra lines in my recipes all the time with no issue. What does the shebang generally resolve to on what operating systems?

**Comment by MostHated** [comment]
My apologies, I didn't realize there was a response. I believe it was PowerShell on Linux (pop_os), but I will have to double check, I jump between linux and a windows vm a lot to test things. 

The issue at the time was locally declared variables at the top of the scope were not seen when I attempted to use them a few lines down. I will attempt to confirm this, though. It had felt like it was ran without the shebang line being present, so each line was being ran individually.
It might have just been a fluke, though, or something specific I had done wrong maybe, because I am actually using a powershell shebang right now, and it has been working properly. So it might be a non-issue. I will try to locate that same script I was using and confirm.
